<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SequenceView</class>
 <widget class="QMainWindow" name="SequenceView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1020</width>
    <height>694</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Cabin</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Primer Designer - Sequence View</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../resources/AlignShop.qrc">
    <normaloff>:/icons/logo/align_shop_logo_256.png</normaloff>:/icons/logo/align_shop_logo_256.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow, QDialog, #body, #centralWidget, QScrollArea {
	background: #FFF;
}

QGroupBox, QFrame {
	border: 0;
}

QSplitter::handle {
	background: #EEE;
}
	
#mainHeader {
	background-color: QLinearGradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #F7FAFE, stop: 0.5 #DCE7F5, stop: 1 #DCE7F5);
	color: #15428B;
	border-bottom: 1px solid #A0AFC3;
    font-size: 16pt;
    padding-top: 10px;
	padding-bottom: 10px;
}

#body {
	margin-top: -5px;
	padding-top: 5px;
}

#sideNav {
 	margin-left: 5px;
	margin-top: 5px;
   	padding-right: 5px;
}

#mainFrame {
	background: #FFF;
	border: 0;
 	margin: 5px;
}

#errorMessageGroup {
	background: #FFF3E8;
	border-bottom: 1px solid #FFA775;
}

PrimerListPanel, #primerList {
	margin-top: 5px;
    padding-top: 5px;
}

QDialogButtonBox {
	padding-right: 10px;
	background: red;
}

QDialogButtonBox QPushButton {
	width: 50px;
}

QStatusBar {
	background-color: QLinearGradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #F7FAFE, stop: 0.5 #DCE7F5, stop: 1 #DCE7F5);
   	color: #15428B;
	font-size: 14pt;
	border: 1px solid #8DB2E3;
	margin-top: 5px;
}

QTextEdit, QLineEdit[frame=&quot;true&quot;], SequenceTextView, QSpinBox, QDoubleSpinBox {
	border-radius: 3px;
	selection-background-color: #CBE1FC;
	selection-color: #000;
	border: 1px solid #E3E9EF;
	border-top-color: #ABADB3;
}

QTextEdit:hover, QTextEdit:focus,  QLineEdit[frame=&quot;true&quot;]:hover,  QLineEdit[frame=&quot;true&quot;]:focus, SequenceTextView:hover, SequenceTextView:focus, QSpinBox:hover, QSpinBox:focus {
	border: 1px solid #C7E2F1;
	border-top-color: #5794BF;
}

/* Wizards */
#PrimerDesignWizard {
	background: #F7FAFE;	
}

QWizardPage {
	background: #FFF;
}

QWizardPage QPushButton {
	min-width:  30;
}

#primerParameterBlock {
	background: #FFF;
}

#ampliconGroupBox {
	border-right: 1px solid #EEE;
}

/* Buttons */
QPushButton {
	color: #000;
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #FCFCFC, stop: 0.1 #F2F2F2, stop: 0.49 #EBEBEB, stop: 0.5 #DDDDDD, stop: 0.9 #CFCFCF, stop: 1 #F3F3F3);
	border: 1px solid #707070;
	border-radius: 3;
	padding: 3px;
	font-size: 10pt;
	padding-left: 5px;
	padding-right: 5px;
	min-width: 50px;
	min-height: 16px;
}

QPushButton:focus
{
    background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #3ACDF2, stop: 0.1 #F2F2F2, stop: 0.49 #E8EFF3, stop: 0.5 #D4E3EA, stop: 0.9 #C4D7E2, stop: 1 #3ACDF2);
    border: 1px solid #3C7FB1;  
}

QPushButton[flat=&quot;false&quot;]:hover {
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #FAFDFE, stop: 0.1 #EAF6FD, stop: 0.49 #D9F0FC, stop: 0.5 #BEE6FD, stop: 0.9 #A7D9F5, stop: 1 #FAFDFE);
  	border-color: #3C7FB1;
}

QPushButton[flat=&quot;false&quot;]:pressed {
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #FAFDFE, stop: 0.1 #EAF6FD, stop: 0.49 #C4E5F6, stop: 0.5 #98D1EF, stop: 0.9 #68B3DB, stop: 1 #9ABACB);
	border-color: #2C628B;
}

/* Hack: flat = toolbar button behavior/appearance. */
QPushButton[flat=&quot;true&quot;] {
	background-color: rgba(255, 255, 255, 0);
	border: 0;
	min-width: 16px;
}

QPushButton[flat=&quot;true&quot;]:disabled {
	background: none;
}

QPushButton[flat=&quot;true&quot;]:hover, QPushButton[flat=&quot;true&quot;]:pressed {
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #FFFCF2, stop: 0.1 #FFFCF2, stop: 0.49 #FFF3CF, stop: 0.5 #FFECB5, stop: 1 #FFECB5);
	border: 1px solid #E5C365;
}

QPushButton[flat=&quot;true&quot;]:pressed {
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #FFECB5, stop: 0.1 #FFECB5, stop: 0.49 #FFECB5, stop: 0.5 #FFF3CF, stop: 1 #FFFCF2);
}

QPushButton:disabled {
	background-color: #F0F0F0;
	color: #AAA;
	border-color: #AAA;
}

/* Tables */
QTableView {
	selection-background-color: #FFF;
}

QTableView::item {
	border: 0px;
	padding: 5px;
}

QTableView::item:selected {
	background: QLinearGradient(x1: 0, y1: 0, X2: 0, y2: 1, stop: 0 #EBF4FD, stop: 0.1 #DCEBFC, stop: 0.9 #C1DBFC, stop: 1 #EBF4FD);
	border-top: 1px solid #7DA2CE;
	border-bottom: 1px solid #7DA2CE;
	color: #000;
}

QHeaderView {
	background: #FFF;
	border: 0;
	font-weight: normal;
}

QHeaderView::down-arrow, QHeaderView::up-arrow {
	image: url(:/icons/odd_size/sort_descending.png);
	width: 7px;
	height: 9px;
	subcontrol-position: top center;
}

QHeaderView::up-arrow {
	image: url(:/icons/odd_size/sort_ascending.png);
}

QHeaderView::section {
	color: #4C607A;
	font-weight: normal;
	padding: 5px;
	background: #FFF;
	border: 0;
	border-right: 1px solid QLinearGradient(x1: 0, y1: 0, X2: 0, y2: 1, stop: 0 #DFEAF7, stop: 0.5 #DFEAF7, stop: 1 #FFF);
}

/* Scroll Regions */
QScrollArea &gt; * &gt; * {
	background-color: #FFF;
}

QScrollBar {
	border: 0;
	background: #FFF;
	min-width: 18px;
	margin: 16px 0;
}

QScrollBar:horizontal {
	min-height: 18px;
	margin: 0 16px;
}

QScrollBar::handle, QScrollBar::add-line, QScrollBar::sub-line {
	border: 1px solid #979797;
	background: QLinearGradient(x1: 0, y1: 0, x2: 1, y2: 0, stop: 0 #F5F5F5, stop: 0.49 #E9E9EB, stop: 0.5 #E9E9EB, stop: 1 #C7C7CB);
	border-radius: 2px;
	min-height: 16px;
	min-width: 16px;
}

QScrollBar::handle:horizontal, QScrollBar::add-line:horizontal, QScrollBar::sub-line:horizontal {
	background: QLinearGradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #F5F5F5, stop: 0.49 #E9E9EB, stop: 0.5 #E9E9EB, stop: 1 #C7C7CB);
}

QScrollBar::handle:horizontal:hover, QScrollBar::add-line:horizontal:hover, QScrollBar::sub-line:horizontal:hover  {
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #E8F1F5, stop: 0.1 #E8F1F5, stop: 0.49 #E8F1F5, stop: 0.5 #BFDBFF, stop: 1 #BFDBFF);
	border: 1px solid #3C7FB1;
}

QScrollBar::handle:vertical:hover, QScrollBar::add-line:vertical:hover, QScrollBar::sub-line:vertical:hover  {
	background-color: QLinearGradient( x1: 0, y1: 0, x2: 1, y2: 0, stop: 0 #E8F1F5, stop: 0.1 #E8F1F5, stop: 0.49 #E8F1F5, stop: 0.5 #BFDBFF, stop: 1 #BFDBFF);
	border: 1px solid #3C7FB1;
}

QScrollBar::add-line, QScrollBar::sub-line {
	height: 16px;
	width: 16px;
	subcontrol-origin: margin;
}

QScrollBar::add-line:horizontal {
	image: url(:/icons/16x16/actions/scroll_right.png);
	subcontrol-position: right;
}

QScrollBar::add-line:vertical {
	image: url(:/icons/16x16/actions/scroll_down.png);
	subcontrol-position: bottom;
}

QScrollBar::sub-line:horizontal {
	image: url(:/icons/16x16/actions/scroll_left.png);
	subcontrol-position: left;
}

QScrollBar::sub-line:vertical {
	image: url(:/icons/16x16/actions/scroll_up.png);
	subcontrol-position: top;
}

QScrollBar::up-arrow:vertical, QScrollBar::down-arrow:vertical {
	border:0;
}

QScrollBar::add-page, QScrollBar::sub-page {
	background: #979797;
}

/* QSpinBox */
 QSpinBox, QDoubleSpinBox {
     padding-right: 15px; /* make room for the arrows */
 }

 QSpinBox::up-button, QDoubleSpinBox::up-button {
     subcontrol-origin: border;
     subcontrol-position: top right; /* position at the top right corner */

     width: 16px; /* 16 + 2*1px border-width = 15px padding + 3px parent border */
     border-width: 1px;
 }

 QSpinBox::up-arrow, QDoubleSpinBox::up-arrow {
     image: url(:/icons/16x16/actions/scroll_up.png);
     width: 16px;
     height: 16px;
 }

 QSpinBox::up-arrow:disabled, QSpinBox::up-arrow:off,
 QDoubleSpinBox::up-arrow:disabled, QDoubleSpinBox::up-arrow:off { /* off state when value is max */
    image: none;
 }

 QSpinBox::down-button, QDoubleSpinBox::down-button {
     subcontrol-origin: border;
     subcontrol-position: bottom right; /* position at bottom right corner */

     width: 16px;
     border-width: 1px;
     border-top-width: 0;
 }
 
 QSpinBox::down-arrow, QDoubleSpinBox::down-arrow {
     image: url(:/icons/16x16/actions/scroll_down.png);
     width: 16px;
     height: 16px;
 }

 QSpinBox::down-arrow:disabled,
 QSpinBox::down-arrow:off, 
 QDoubleSpinBox::down-arrow:disabled, 
 QDoubleSpinBox::down-arrow:off { 
    image: none;
 }</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="body">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="handleWidth">
       <number>2</number>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QFrame" name="sideNav">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QFrame" name="dnaSequencesHeaderGroup">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>100</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="HeaderLabel" name="dnaSequencesHeader">
             <property name="font">
              <font>
               <family>Cabin</family>
               <pointsize>14</pointsize>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="text">
              <string>DNA Sequences</string>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>16</height>
              </size>
             </property>
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton">
             <property name="minimumSize">
              <size>
               <width>26</width>
               <height>22</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>24</width>
               <height>24</height>
              </size>
             </property>
             <property name="font">
              <font>
               <family>Cabin</family>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="toolTip">
              <string>Add New Sequence</string>
             </property>
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../../resources/AlignShop.qrc">
               <normaloff>:/icons/16x16/actions/add.png</normaloff>:/icons/16x16/actions/add.png</iconset>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="dnaSequenceFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QListView" name="listView">
             <property name="font">
              <font>
               <family>Cabin</family>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="showDropIndicator" stdset="0">
              <bool>false</bool>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="mainFrame">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>10</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>Cabin</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWidth">
        <number>1</number>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="SequenceDetailsPanel" name="sequenceDetails">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="zeroItemsFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="zeroItemsLabel">
             <property name="font">
              <font>
               <family>Cabin</family>
               <pointsize>16</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Please select a sequence, or click the plus icon to create a new sequence.</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1020</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="action_Save"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SequenceDetailsPanel</class>
   <extends>QFrame</extends>
   <header>forms/PrimerDesign/SequenceDetailsPanel.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HeaderLabel</class>
   <extends>QLabel</extends>
   <header>forms/PrimerDesign/Labels.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources/AlignShop.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>SequenceView</receiver>
   <slot>onAddSequenceClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>563</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>582</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>SequenceView</receiver>
   <slot>onFileOpenClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Save</sender>
   <signal>triggered()</signal>
   <receiver>SequenceView</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>SequenceView</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>SequenceView</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onAddSequenceClicked()</slot>
  <slot>onSelectedItemChanged(QModelIndex)</slot>
  <slot>onSequenceNameValueEdited()</slot>
  <slot>onListItemPressed(QModelIndex)</slot>
  <slot>onSequenceNotesValueEdited()</slot>
  <slot>onFileOpenClicked()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
 </slots>
</ui>
